rdflib/plugins/serializers/turtle.py

```diff
@@ -284,6 +284,26 @@ def getQName(self, uri, gen_prefix=True):
 
             if pfx is not None:
                 parts = (pfx, uri, "")
+
```

Comment:
Ideally compute_qname (which is incorrectly named) should return the PrefixedName in components, and then all that should have to happen here is some standard escaping of the PN_LOCAL.

Not sure why this is not the case, I will debug a bit. Having compute_qname do what it should do is preferable to having similar logic in here.

Reply:
It will also make it easier to fix for other serializers, such as TriG and N3.

Reply:
The actual exception here is:

```
2022-07-22T01:23:42.590 DEBUG    root         turtle.py:283:getQName caught ...
Traceback (most recent call last):
  File "/home/iwana/sw/d/github.com/iafork/rdflib.reviews/rdflib/plugins/serializers/turtle.py", line 280, in getQName
    parts = self.store.compute_qname(uri, generate=gen_prefix)
  File "/home/iwana/sw/d/github.com/iafork/rdflib.reviews/rdflib/graph.py", line 1057, in compute_qname
    return self.namespace_manager.compute_qname(uri, generate)
  File "/home/iwana/sw/d/github.com/iafork/rdflib.reviews/rdflib/namespace/__init__.py", line 522, in compute_qname
    raise KeyError(
KeyError: 'No known prefix for http://www.example.com/foo/obj# and generate=False'
```

Which makes me think the real issue is not in this function but elsewhere.